code/game/turfs/open/floor.dm

```diff
@@ -61,21 +61,22 @@
 	return ..()
 
 /turf/open/floor/ex_act(severity, target)
-	var/shielded = is_shielded()
-	..()
-	if(severity != 1 && shielded && target != src)
-		return
+	. = ..()
+
 	if(target == src)
 		ScrapeAway(flags = CHANGETURF_INHERIT_AIR)
-		return
-	if(target != null)
-		severity = 3
+		return TRUE
+	if(severity != EXPLODE_DEVASTATE && is_shielded())
+		return FALSE
```

Comment:
Why swap this around?

Reply:
I saw a chance to remove some redundant checks. The old way checked whether the explosion was targeting the floor twice and cached a variable that was only used once.

Reply:
Doesn't this mean that the scrapaway can happen even if the turf is strong or the explosion weak?

Reply:
Yes. But it it could do that already. Notice the `target != src` check in the original early return.

Reply:
If the explosion was targeting the floor it would automatically bypass the original early return and scrapeaway the turf.

Reply:
ah excuse me